tests/phpunit/tests/export/attachments.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @group export
+ * @group attachments
+ * @ticket 17379
+ */
+class Test_Export_Includes_Attachments extends WP_UnitTestCase {
+	protected static $post_1;
+	protected static $attachment_1;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$post_1 = self::factory()->post->create_and_get(
+			array(
+				'post_title'  => 'Test Post 1',
+				'post_author' => $factory->user->create( array( 'role' => 'editor' ) ),
+			)
+		);
+
+		$post_2 = self::factory()->post->create_and_get(
+			array(
+				'post_title'  => 'Test Post 2',
+				'post_author' => $factory->user->create( array( 'role' => 'editor' ) ),
+			)
+		);
+
+		$file = DIR_TESTDATA . '/images/test-image.jpg';
+
+		self::$attachment_1 = $factory->attachment->create_upload_object( $file );
+
+		set_post_thumbnail( self::$post_1->ID, self::$attachment_1 );
+		set_post_thumbnail( $post_2, $factory->attachment->create_upload_object( $file ) );
+	}
+
+	/**
+	 * Tests the export function to ensure that attachments are included.
+	 *
+	 * Runs in a separate process to prevent "headers already sent" error.
+	 *
+	 * This test does not preserve global state to prevent the exception
+	 * "Serialization of 'Closure' is not allowed" when running in
+	 * a separate process.
+	 *
+	 * @runInSeparateProcess
+	 * @preserveGlobalState disabled
+	 */
+	public function test_export_includes_attachments_for_specific_author() {
```

Comment:
It would be great to have more test coverage to cover the other selectable states such as:

* Selecting to export all posts but not selecting a specific author.
* Selecting to export all pages but not selecting a specific author.
* Selecting to export all pages from a specific author.

Reply:
We now have the following unit tests:

1. When exporting all content, all posts and pages are exported as well as their attachments
2. When exporting posts, all posts are exported along with their attachments
3. When exporting pages, all pages are exported along with their attachments
4. When exporting pages by a specific author, their pages and attachments are exported
5. When exporting posts by a specific author, their posts and attachments are exported